### PR TITLE
Super 16 Puzzle Game

### PR DESCRIPTION
**Pull request by swreinehr**
## Description

This is a keymap that implements the 15/16 puzzle game with the rgb leds on a super 16. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/super16/keymaps/15game/config.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2020 Sam Reinehr
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/config.h]: This file can be deleted.

File: keyboards/1upkeyboards/super16/keymaps/15game/keymap.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    K10,
+    K11,
+    K12,
+    K13,
+    K14,
+    K15,
+};
+/* just a simple way to give each key a unique code */
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {[0] = LAYOUT_ortho_4x4(/* Base */
+                                                                                     K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K10, K11, K12, K13, K14, K15)};
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]:
```suggestion
// clang-format off
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT_ortho_4x4(/* Base */
        K00, K01, K02, K03,
        K04, K05, K06, K07,
        K08, K09, K10, K11,
        K12, K13, K14, K15
    )
};
// clang-format on
```

Comment (swreinehr) [reply keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]: I've made a few more formatting changes to make the array structure present in all the maps

File: keyboards/1upkeyboards/super16/keymaps/15game/keymap.c [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    K14,
+    K15,
+};
+/* just a simple way to give each key a unique code */
+//clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_4x4(/* Base */                                                   K00, K01, K02, K03,
+	K04, K05, K06, K07,
+	K08, K09, K10, K11,
+	K12, K13, K14, K15
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]:
```suggestion
    /* Base */
    [0] = LAYOUT_ortho_4x4(
        K00, K01, K02, K03,
        K04, K05, K06, K07,
        K08, K09, K10, K11,
        K12, K13, K14, K15
```

File: keyboards/1upkeyboards/super16/keymaps/15game/keymap.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+};
+/* flags describing current free square/0 */
+uint8_t current = 0;
+/* r g and b describe the colors for the initial map,
+currently blank for free, and evenly spaced hues with maximum sat/value  */
+const uint8_t r[16] = {
+   0x00, 0xFF, 0xFF, 0xFF,
+   0xCC, 0x66, 0x00, 0x00,
+   0x00, 0x00, 0x00, 0x00,
+   0x66, 0xCC, 0xFF, 0xFF
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]:
```suggestion
    0x00, 0xFF, 0xFF, 0xFF,
    0xCC, 0x66, 0x00, 0x00,
    0x00, 0x00, 0x00, 0x00,
    0x66, 0xCC, 0xFF, 0xFF
```

File: keyboards/1upkeyboards/super16/keymaps/15game/keymap.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+};
+const uint8_t g[16] = {
+   0x00, 0x00, 0x66, 0xCC,
+   0xFF, 0xFF, 0xFF, 0xFF,
+   0xFF, 0xCC, 0x66, 0x00,
+   0x00, 0x00, 0x00, 0x00
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]:
```suggestion
    0x00, 0x00, 0x66, 0xCC,
    0xFF, 0xFF, 0xFF, 0xFF,
    0xFF, 0xCC, 0x66, 0x00,
    0x00, 0x00, 0x00, 0x00
```

File: keyboards/1upkeyboards/super16/keymaps/15game/keymap.c [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+};
+const uint8_t b[16] = {
+   0x00, 0x00, 0x00, 0x00,
+   0x00, 0x00, 0x00, 0x66,
+   0xCC, 0xFF, 0xFF, 0xFF,
+   0xFF, 0xFF, 0xCC, 0x66
```
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/keymaps/15game/keymap.c]:
```suggestion
    0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0x00, 0x66,
    0xCC, 0xFF, 0xFF, 0xFF,
    0xFF, 0xFF, 0xCC, 0x66
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**